core/src/main/java/org/apache/iceberg/BaseIncrementalScan.java

```diff
@@ -34,6 +34,17 @@ protected BaseIncrementalScan(Table table, Schema schema, TableScanContext conte
   protected abstract CloseableIterable<T> doPlanFiles(
       Long fromSnapshotIdExclusive, long toSnapshotIdInclusive);
 
+  @Override
+  public ThisT fromSnapshotInclusive(String ref) {
+    SnapshotRef snapshotRef = table().refs().get(ref);
+    Preconditions.checkArgument(snapshotRef != null, "Cannot find ref %s", ref);
+    Preconditions.checkArgument(snapshotRef.isTag(), "Ref %s is not a tag", ref);
+
+    TableScanContext newContext = context().fromSnapshotIdInclusive(snapshotRef.snapshotId());
+
+    return newRefinedScan(table(), schema(), newContext);
```

Comment:
should we rather call `fromSnapshotInclusive(snapshotRef.snapshotId())` here to be consistent with other methods?